### PR TITLE
Log git clone output only on debug/trace log level

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
The clone information is not very useful, especially when printing it
out in Google Cloud Build. Therefore we now disable it and print it only
if the debug log level is enabled.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Disable `git` clone output if log level is not debug or trace.
```

### CODE REVIEW

File: pkg/git/git.go [hunk 1]
```diff
@@ -274,9 +274,17 @@ func CloneOrOpenRepo(repoPath, repoURL string, useSSH bool) (*Repo, error) {
 		targetDir = t
 	}
 
+	var progress *os.File
```
Comment (hasheddan) [inline pkg/git/git.go:277]: I wonder if we should pipe to stderr on normal log-level so we can dump progress logs in case of error

**Review by cpanato — APPROVED** [review]
/lgtm

File: pkg/git/git.go [hunk 2]
```diff
@@ -274,10 +276,27 @@ func CloneOrOpenRepo(repoPath, repoURL string, useSSH bool) (*Repo, error) {
...
-		Progress: os.Stdout,
+		Progress: io.MultiWriter(progressWriters...),
 	}); err != nil {
+		// Print the stack only if not already done
+		if logLevel < logrus.DebugLevel {
+			logrus.Errorf(
+				"Clone repository failed. Tracked progress:\n%s",
+				progressBuffer.String(),
+			)
+		}
```
Comment (saschagrunert) [inline pkg/git/git.go:299]: @hasheddan now we can print the progress if not already done, how is that? :)

**Comment by saschagrunert** [comment]
/assign @cpanato 
for another lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**